nio_command_block.py

```diff
@@ -19,26 +19,25 @@ class SecurityMethod(Enum):
 
 
 class URLParameter(PropertyHolder):
-    prop_name = ExpressionProperty(title="Property Name")
-    prop_value = ExpressionProperty(title="Property Value")
+    prop_name = Property(title="Property Name")
+    prop_value = Property(title="Property Value")
 
 
 class BasicAuthCreds(PropertyHolder):
-    username = StringProperty(title='Username')
-    password = StringProperty(title='Password')
+    username = StringProperty(title='Username', default='Admin')
+    password = StringProperty(title='Password', default='Admin')
 
 
-@Discoverable(DiscoverableType.block)
-class NioCommand(OAuth2, Block):
+class NioCommand(OAuth2ServiceAccount, Block):
 
     params = ListProperty(URLParameter,
                           title="Command Parameters",
                           default=[])
     host = StringProperty(title="n.io Host", default="[[NIOHOST]]")
     port = IntProperty(title="n.io Port", default="[[NIOPORT]]")
-    service_name = ExpressionProperty(title="Service Name")
-    block_name = ExpressionProperty(title="Block Name (optional)")
-    command_name = ExpressionProperty(title="Command Name")
+    service_name = Property(title="Service Name", allow_none=True)
+    block_name = Property(title="Block Name (optional)", allow_none=True)
+    command_name = Property(title="Command Name", allow_none=True)
```

Comment:
I think I know why you did this and it's because nio can be a little weird if you don't either allow none or have a default.

It would probably make more sense for these instead to have a default of empty string. allow_none means that the block property can be `None` (which normally isn't allowed. What would happen in the code if service or command really were None? Your url would probably look pretty funky.